src/common/dns_utils.cpp

```diff
@@ -323,4 +324,106 @@ bool DNSResolver::check_address_syntax(const char *addr) const
   return true;
 }
 
+namespace detail
+{
+  //-----------------------------------------------------------------------
+  // TODO: parse the string in a less stupid way, probably with regex
+  std::string address_from_txt_record(const std::string& s)
+  {
+    // make sure the txt record has "oa1:xmr" and find it
+    auto pos = s.find("oa1:xmr");
+    // search from there to find "recipient_address="
+    pos = s.find("recipient_address=", pos);
```

Comment:
I simply copied the entire function from wallet2.cpp... I'll just add `if (pos == string::npos) return {};` then.